app/controllers/scenarios_controller.rb

```diff
@@ -95,6 +95,17 @@ def update
     end
   end
 
+  def enable_or_disable_all_agents
+    @scenario = current_user.scenarios.find(params[:id])
+
+    respond_to do |format|
+      if @scenario.agents.update_all(disabled: params[:scenario][:disabled] == 'true')
```

Comment:
I'd suggest just:

```ruby
@scenario.agents.update_all(disabled: params[:scenario][:disabled] == 'true')
respond_to do |format|
  format.html { redirect_to @scenario, notice: 'The agents in this scenario have been successfully updated.' }
  format.json { head :no_content }
end
```

No `if`, since the return value of `update_all` isn't meaningful.